### PR TITLE
feat: Implement ML Feature Details API

### PR DESCRIPTION
**Pull request by allisonsuarez**
### Summary of Changes

Implemented Feature Details API and added unit tests
### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by verdan** [comment]
@allisonsuarez please make sure to rebase from the latest master. This will fix your frontend build.

**Comment by allisonsuarez** [comment]
> @allisonsuarez please make sure to rebase from the latest master. This will fix your frontend build.

thanks Verdan! I'll wait until the branch this works stems from is merged and do that :)

File: metadata/metadata_service/proxy/neo4j_proxy.py
Comment (allisonsuarez) [reply metadata/metadata_service/proxy/neo4j_proxy.py:1720]: They are but that also means more calls to neo4j rather than a single call

Comment (sewardgw) [reply metadata/metadata_service/proxy/neo4j_proxy.py:1720]: I should have clarified, for example, this code exists currently in main and is used for table metadata 
```python
        for record in wmk_records:
            if record['key'] is not None:
                watermark_type = record['key'].split('/')[-2]
                wmk_result = Watermark(watermark_type=watermark_type,
                                       partition_key=record['partition_key'],
                                       partition_value=record['partition_value'],
                                       create_time=record['create_time'])
                wmk_results.append(wmk_result)
```
Could this be extracted to a function akin to the following signature?
```python
def get_wmks(input_values: List[Dict]) -> List[Watermark]:
    ...

```

File: frontend/requirements.txt
Comment (mgorsk1) [inline frontend/requirements.txt:70]: maybe it's woth to keep the deps 1:1 with other services? metadata already expects 0.11.0

File: metadata/metadata_service/proxy/neo4j_proxy.py
Comment (allisonsuarez) [reply metadata/metadata_service/proxy/neo4j_proxy.py:1720]: oooh I see yeah I can do that

File: frontend/requirements.txt
Comment (verdan) [reply frontend/requirements.txt]: https://github.com/amundsen-io/amundsen/pull/1088

**Review by danwom — APPROVED** [review]

**Pull request merged**
